AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m

```diff
@@ -80,6 +80,10 @@ - (void)setWrapperSdk:(MSWrapperSdk *)wrapperSdk {
   @synchronized(self) {
     wrapperSdkInformation = wrapperSdk;
     needRefresh = YES;
+
+    // Replace the last device without wrapperSdk in the UserDefaults with an updated info.
+    [self.deviceHistory removeLastObject];
```

Comment:
Do you need this line?

Reply:
Yes, because at the moment the last device in the history does not contain wrapperSdk. But when calling [self device] we save all items stored in deviceHistory.

Reply:
What I meant was that you don't need to delete the last one. This will lead to pick a wrong device info for logs that are created before this line instead of device info that doesn't have wrapper SDK info. Both are not good but wrong device info could be worse.